spec/rester/utils/rspec_utils_spec.rb

```diff
@@ -0,0 +1,274 @@
+                }
+              }
+            }
+
+            it 'should raise an error' do
+              expect { subject }.to raise_error Errors::StubError, "Stub[:some_hash][:some_key]=:some_value doesn't match Subject[:some_hash][:some_key]=:a_different_value"
+            end
+          end # with different hash value
+
+          context 'with different array element' do
+            let(:subj) {
+              {
+                this: :that,
+                some_array: [1, 2, 4]
+              }
+            }
+            let(:stub) {
+              {
+                this: :that,
+                some_array: [1, 2, 3]
+              }
+            }
+
+            it 'should raise an error' do
+              expect { subject }.to raise_error Errors::StubError, "Stub[:some_array][2]=3 doesn't match Subject[:some_array][2]=4"
+            end
+          end # with different array element
+        end # with failures
+
+        context 'with successes' do
+          context 'with simple hash' do
+            let(:subj) {
+              {
+                integer: 1,
+                float: 1.23,
+                string: 'hello',
+                symbol: :hello,
+                boolean: true,
+                another_boolean: false,
+                extra_string: 'extra',
+                extra_symbol: :extra
+              }
+            }
+            let(:stub) {
+              {
+                integer: 1,
+                float: 1.23,
+                string: 'hello',
+                symbol: :hello,
+                boolean: true,
+                another_boolean: false
+              }
+            }
+
+            it { is_expected.to be true }
+          end # with simple hash
+
+          context 'with nested hash' do
+            let(:subj) {
+              {
+                this: :that,
+                some_hash: {
+                  integer: 1,
+                  float: 1.23,
+                  string: 'hello',
+                  symbol: :hello,
+                  boolean: true,
+                  another_boolean: false,
+                  extra_string: 'extra',
+                  extra_symbol: :extra
+                }
+              }
+            }
+            let(:stub) {
+              {
+                this: :that,
+                some_hash: {
+                  integer: 1,
+                  float: 1.23,
+                  string: 'hello',
+                  symbol: :hello,
+                  boolean: true,
+                  another_boolean: false,
+                  extra_string: 'extra',
+                  extra_symbol: :extra
+                }
+              }
+            }
+
+            it { is_expected.to be true }
+          end # with nested hash
+
+          context 'with nested array' do
+            let(:subj) {
+              {
+                this: :that,
+                some_array: [1, 2, 3],
+                extra_field: 1
+              }
+            }
```

Comment:
Thoughts on making the test more focused?  If we're testing that it will match a nested array properly, why not just `let(:subj) { { some_array: [1,2,3] } }`?